### PR TITLE
Removed redefinition of VarMapMal and moved it to VarMapVal.h. 

### PR DESCRIPTION
**Pull request by Tonyx97**
This change also makes the project perfectly compilable for C++20, everything works just fine.

### CODE REVIEW

**Review by Adngel — APPROVED** [review]

**Review by hispidence — APPROVED** [review]

File: TombEngine/Scripting/Include/ScriptInterfaceGame.h [hunk 1]
```diff
@@ -6,16 +6,9 @@
 #include "Game/room.h"
 #include "Specific/level.h"
 
+#include "VarMapVal.h"
```
Comment (Sezzary) [inline TombEngine/Scripting/Include/ScriptInterfaceGame.h:9]: Remember to use full paths. Our conventions prefer them. :)

File: TombEngine/Scripting/Include/VarMapVal.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+using VarMapVal = std::variant<
+	short,
+	std::reference_wrapper<MESH_INFO>,
+	std::reference_wrapper<LevelCameraInfo>,
+	std::reference_wrapper<SinkInfo>,
+	std::reference_wrapper<SoundSourceInfo>,
+	std::reference_wrapper<TriggerVolume>,
+	std::reference_wrapper<AI_OBJECT>,
+	std::reference_wrapper<ROOM_INFO>>;
```
Comment (Sezzary) [inline TombEngine/Scripting/Include/VarMapVal.h:11]: Remember newline at end of file.

File: TombEngine/Specific/level.cpp [hunk 3]
```diff
@@ -184,9 +184,9 @@ void LoadItems()
 			item->TriggerFlags = ReadInt16();
 			item->Flags = ReadInt16();
 			item->Name = ReadString();
-
-			g_GameScriptEntities->AddName(item->Name, i);
-			g_GameScriptEntities->TryAddColliding(i);
+			
+			g_GameScriptEntities->AddName(item->Name, static_cast<short>(i));
```
Comment (Sezzary) [inline TombEngine/Specific/level.cpp:188]: Conventions prefer C-style casts unless `static_cast` is specifically required. Is it required here?

Comment (Tonyx97) [reply TombEngine/Specific/level.cpp:188]: Not really. I forgot to remove it after reading the guidelines yesterday.

**Pull request merged**